nsqdserver/protocol_v2.go

```diff
@@ -559,18 +571,21 @@ func (p *protocolV2) messagePump(client *nsqd.ClientV2, startedChan chan bool,
 				subChannel.ContinueConsumeForOrder()
 				continue
 			}
-
 			shouldSend, err := subChannel.StartInFlightTimeout(msg, client, client.String(), msgTimeout)
 			if !shouldSend || err != nil {
 				continue
 			}
+
 			lastActiveTime = time.Now()
 			client.SendingMessage()
 			err = SendMessage(client, msg, &buf, subChannel.IsOrdered())
 			if err != nil {
+				nsqd.NsqLogger().Errorf("fail to send message %v order %v", err, subChannel.IsOrdered())
 				goto exit
 			}
 			flushed = false
+		case <-client.TagUpdateNotifyChan:
```

Comment:
What this for? I think a client connection should never allowed to change the its tag